homeassistant/components/verisure/config_flow.py

```diff
@@ -0,0 +1,173 @@
+    ResponseError as VerisureResponseError,
+    Session as Verisure,
+)
+import voluptuous as vol
+
+from homeassistant.config_entries import (
+    CONN_CLASS_CLOUD_POLL,
+    ConfigEntry,
+    ConfigFlow,
+    OptionsFlow,
+)
+from homeassistant.const import CONF_EMAIL, CONF_PASSWORD
+from homeassistant.core import callback
+
+from .const import (  # pylint:disable=unused-import
+    CONF_GIID,
+    CONF_LOCK_CODE_DIGITS,
+    CONF_LOCK_DEFAULT_CODE,
+    DEFAULT_LOCK_CODE_DIGITS,
+    DOMAIN,
+    LOGGER,
+)
+
+
+class VerisureConfigFlowHandler(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Verisure."""
+
+    VERSION = 1
+    CONNECTION_CLASS = CONN_CLASS_CLOUD_POLL
+
+    installations: dict[str, str]
+    email: str
+    password: str
+
+    # These can be removed after YAML import has been removed.
+    giid: str | None = None
+    settings: dict[str, int | str] = {}
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(entry: ConfigEntry):
+        """Get the options flow for this handler."""
+        return VerisureOptionsFlowHandler(entry)
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> dict[str, Any]:
+        """Handle the initial step."""
+        errors: dict[str, str] = {}
+
+        if user_input is not None:
+            verisure = Verisure(
+                username=user_input[CONF_EMAIL], password=user_input[CONF_PASSWORD]
+            )
+            try:
+                await self.hass.async_add_executor_job(verisure.login)
+            except VerisureLoginError as ex:
+                LOGGER.debug("Could not log in to Verisure, %s", ex)
+                errors["base"] = "invalid_auth"
+            except (VerisureError, VerisureResponseError) as ex:
+                LOGGER.debug("Unexpected response from Verisure, %s", ex)
+                errors["base"] = "unknown"
+            else:
+                self.email = user_input[CONF_EMAIL]
+                self.password = user_input[CONF_PASSWORD]
+                self.installations = {
+                    inst["giid"]: f"{inst['alias']} ({inst['street']})"
+                    for inst in verisure.installations
+                }
+
+                return await self.async_step_installation()
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_EMAIL): str,
+                    vol.Required(CONF_PASSWORD): str,
+                }
+            ),
+            errors=errors,
+        )
+
+    async def async_step_installation(
+        self, user_input: dict[str, Any] | None = None
+    ) -> dict[str, Any]:
+        """Select Verisure installation to add."""
+        if len(self.installations) == 1:
+            user_input = {CONF_GIID: list(self.installations)[0]}
+        elif self.giid and self.giid in self.installations:
+            user_input = {CONF_GIID: self.giid}
```

Comment:
wouldn't this be better to do in step_user and input the user_input to step_installation?

Reply:
It doesn't matter that much 🤷

```py
        elif self.giid and self.giid in self.installations:
            user_input = {CONF_GIID: self.giid}
```

Will be removed after 2 releases. So that leaves 1 "guard" in this method after two releases. I think that is fine?